### PR TITLE
UCP/WIREUP: do not use UCP_EP_FLAG_CONNECT_REQ_QUEUED for CM case

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
do not use UCP_EP_FLAG_CONNECT_REQ_QUEUED for CM case

## Why ?
this prevents resources leak when disconnecting of not completely
connected UCP EPs

### CODE REVIEW

File: src/ucp/core/ucp_ep.c [hunk 1]
```diff
@@ -334,7 +334,9 @@ ucs_status_t ucp_ep_init_create_wireup(ucp_ep_h ep, unsigned ep_init_flags,
     }
 
     ep->am_lane = key.am_lane;
-    ep->flags  |= UCP_EP_FLAG_CONNECT_REQ_QUEUED;
+    if (ucp_ep_get_cm_lane(ep) == UCP_NULL_LANE) {
```
Comment (yosefe) [inline src/ucp/core/ucp_ep.c:337]: reuse https://github.com/openucx/ucx/pull/5220/files#diff-6ec7d6c877f4e5aae320aa49bd08f22cR241

File: src/ucp/wireup/wireup.c [hunk 2]
```diff
@@ -642,10 +642,16 @@ static ucs_status_t ucp_wireup_msg_handler(void *arg, void *data,
...
     ucp_unpacked_address_t remote_address;
+    ucp_ep_h ep UCS_V_UNUSED;
     ucs_status_t status;
 
     UCS_ASYNC_BLOCK(&worker->async);
 
+    if (msg->dest_ep_ptr != 0) {
+        ep = ucp_worker_get_ep_by_ptr(worker, msg->dest_ep_ptr);
+        ucs_assert(ucp_ep_get_cm_lane(ep) == UCP_NULL_LANE);
+    }
```
Comment (yosefe) [inline src/ucp/wireup/wireup.c:653]: why needed? we're planning to expand wireup for CM case as well to send extra transport info

File: src/ucp/core/ucp_ep.c (hunk 1 above)
Comment (evgeny-leksikov) [reply src/ucp/core/ucp_ep.c:337]: ok, will create another PR (as you sugested) and reuse it in other ones

File: src/ucp/wireup/wireup.c (hunk 2 above)
Comment (evgeny-leksikov) [reply src/ucp/wireup/wireup.c:653]: but today this flow is impossible, I see 2 reasons for adding:
1) to catch mixing CM/iface client-server connection establishing for current code
2) it's easier to extend protocol getting assertion failures on impossible flows before

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
bot:pipe:retest

File: src/ucp/wireup/wireup.c (hunk 2 above)
Comment (yosefe) [reply src/ucp/wireup/wireup.c]: ok, pls add code comment

Comment (evgeny-leksikov) [reply src/ucp/wireup/wireup.c]: added https://github.com/openucx/ucx/pull/5244/commits/5a2118b4d9871d2c12db6f3eded9ffd4e08a7221#diff-5e97c1d955a563dd705187d5ded3b527R652

**Comment by evgeny-leksikov** [comment]
bot:pipe:retest

**Comment by evgeny-leksikov** [comment]
fuilure is https://github.com/openucx/ucx/issues/5192
bot:pipe:retest

**Comment by evgeny-leksikov** [comment]
azp failure is https://github.com/openucx/ucx/issues/5260

**Pull request merged**